### PR TITLE
Use saturating sub as appropriate for thread-count

### PR DESCRIPTION
**Pull request by Ben-PH**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

* [x] confirm `.get() - 1` only: no subtractions that can potentially underflow

https://github.com/massalabs/massa/pull/3489#discussion_r1142152635

### CODE REVIEW

**Comment by Ben-PH** [comment]
I'll see what I can do with `NonZeroU8`:

**Comment by Ben-PH** [comment]
I've changed thread-count to use `NonZeroU8` wherever I could find `thread_count: u8`.

This introduces one line of unsafe code:

https://github.com/massalabs/massa/pull/3695/files#diff-e151d5d22edfdbbd0bc0fb2bb32a88393e99a976e67ee8be309af16d59816332R89-R91


I'll self review for arithmatic issues

**Review by Ben-PH — COMMENTED** [review]
todo: go through "as usize/u64" and use `.into()` were possible

**Comment by damip** [comment]
Note: doing thread_count.get() -1 is defeating the purpose of the type that explicitly does not implement the "+" and "-" operators for very good reasons. Use saturating_sub or checked_sub.

**Comment by damip** [comment]
This is a NON-PRIORITARY refactoring

File: massa-models/src/slot.rs [hunk 1]
```diff
@@ -199,10 +200,10 @@ impl Slot {
     }
 
     /// returns the maximal slot
-    pub const fn max(thread_count: u8) -> Slot {
+    pub const fn max(thread_count: NonZeroU8) -> Slot {
         Slot {
             period: u64::MAX,
-            thread: thread_count,
+            thread: thread_count.get() - 1,
```
Comment (damip) [inline massa-models/src/slot.rs:206]: this is the only PRIORITARY change

**Pull request closed without merging**